dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -463,148 +467,155 @@ private LambdaEvent createBasicEvent(String repository, String gitReference, Str
         return lambdaEvent;
     }
 
+    private void checkDuplicateNames(List<? extends Workflowish> workflows, List<? extends Workflowish> tools) {
```

Comment:
Should services also be checked too? A service has no name, so its name is `""`, and in this function, you're checking that only one entry has no name.

Reply:
I think it's theoretically possible to have a service with the same name as a workflow/apptool, and in the past, we implemented a test to confirm that possibility.  But, chatting with Denis today, he mentioned that maybe it was a good idea that services _not_ be able to have the same name as a workflow/tool, so I added the service to the check.